node-lib/src/options.rs

```diff
@@ -25,14 +25,17 @@ use std::{
 use clap::{Args, Parser, Subcommand};
 use common::chain::config::{regtest_options::ChainConfigOptions, ChainType};
 use p2p::types::ip_or_socket_address::IpOrSocketAddress;
-use utils::default_data_dir::default_data_dir_common;
+use utils::{clap_utils, default_data_dir::default_data_dir_common};
 
 use crate::config_files::{NodeTypeConfigFile, StorageBackendConfigFile};
 
 const CONFIG_NAME: &str = "config.toml";
 
 /// Mintlayer node executable
+// Note: this struct is shared between different node executables, namely, node-daemon and node-gui,
+// so the env vars for both of them will use the same infix; this is intended.
 #[derive(Parser, Debug)]
+#[clap(mut_args(clap_utils::env_adder("NODE")))]
 #[clap(author, version, about)]
 pub struct Options {
```

Comment:
I turned out that this doesn't work for subcommands. Instead, I have to specify it for each subcommand individually as well:
```
#[derive(Subcommand, Clone, Debug)]
pub enum Command {
    /// Run the mainnet node.
    #[clap(mut_args(clap_utils::env_adder("NODE")))]
    Mainnet(RunOptions),
    /// Run the testnet node.
    #[clap(mut_args(clap_utils::env_adder("NODE")))]
    Testnet(RunOptions),
    /// Run the regtest node.
    #[clap(mut_args(clap_utils::env_adder("NODE")))]
    Regtest(Box<RegtestOptions>),
}
```
Which makes me wonder whether we should use different infixes for different networks then. E.g.
```
#[derive(Subcommand, Clone, Debug)]
pub enum Command {
    /// Run the mainnet node.
    #[clap(mut_args(clap_utils::env_adder("MAINNET_NODE")))]
    Mainnet(RunOptions),
    /// Run the testnet node.
    #[clap(mut_args(clap_utils::env_adder("TESTNET_NODE")))]
    Testnet(RunOptions),
    /// Run the regtest node.
    #[clap(mut_args(clap_utils::env_adder("REGTEST_NODE")))]
    Regtest(Box<RegtestOptions>),
}
```
The env var names will then look like `ML_MAINNET_NODE_P2P_BIND_ADDRESSES`, `ML_TESTNET_NODE_P2P_BIND_ADDRESSES` etc. @TheQuantumPhysicist what do you think?

In the wallet it's a bit different btw. There, the same options are present both on the top-most and subcommand level. So if we do the same for the corresponding subcommand struct in wallet_cli_lib::config:
```
#[derive(Subcommand, Clone, Debug)]
pub enum Network {
    #[clap(mut_args(clap_utils::env_adder("MAINNET_WALLET")))]
    Mainnet(CliArgs),
    #[clap(mut_args(clap_utils::env_adder("TESTNET_WALLET")))]
    Testnet(CliArgs),
    #[clap(mut_args(clap_utils::env_adder("REGTEST_WALLET")))]
    Regtest(Box<RegtestOptions>),
    #[clap(mut_args(clap_utils::env_adder("SIGNET_WALLET")))]
    Signet(CliArgs),
}
```
while still having `#[clap(mut_args(clap_utils::env_adder("WALLET")))]` on `WalletCliArgs`, we'll have both kinds of names. E.g. we'll have `ML_WALLET_NODE_RPC_ADDRESS` for the case when the wallet is run without a subcommand, `ML_TESTNET_WALLET_NODE_RPC_ADDRESS` when it's run with "testnet" etc. But it still makes sense.

P.S. I've just noticed that the infixes that we use sometimes result in some strange env var names. E.g. for the `--wallet-file` option, the env var is named `ML_WALLET_WALLET_FILE`. Perhaps we should use more "schematic" infixes, e.g. `WLT` instead of `WALLET`, so that the var name would become `ML_WLT_WALLET_FILE`.